externals/kyuubi-flink-sql-engine/pom.xml

```diff
@@ -30,6 +30,12 @@
     <name>Kyuubi Project Engine Flink SQL</name>
     <url>https://kyuubi.apache.org/</url>
 
+    <!-- Keep Flink Engine with Scala-2.12. -->
```

Comment:
I know that, currently, the whole Kyuubi project can be compiled with Scala 2.13 except for the Flink engine, while considering that the official Kyuubi releases always use Scala 2.12, we don't need to make such changes.

Please revert Flink engine Scala version related changes.

In summary, what we need to do are:
1. upgrade Scala 2.13.11 in another PR ahead
2. just like what you do in https://github.com/apache/kyuubi/pull/6306, tune the root `pom.xml` and add profile `java-21`
3. enable Java 21 test on CI with Spark 4.0

Reply:
Thank you for your guidance, I will create a new PR to upgrade the Scala version to 2.13.11 soon